src/python/pants/engine/native.py

```diff
@@ -283,6 +283,20 @@ def extern_get_type_for(self, context_handle, val):
     type_id = c.to_id(type(obj))
     return TypeId(type_id)
 
+  @_extern_decl('Handle', ['ExternContext*', 'TypeId'])
+  def extern_get_handle_from_type_id(self, context_handle, ty):
+    c = self._ffi.from_handle(context_handle)
+    obj = c.from_id(ty.tup_0)
+    return c.to_value(obj)
+
+  @_extern_decl('bool', ['ExternContext*', 'TypeId'])
+  def extern_is_union(self, context_handle, type_id):
+    "Return whether or not a type is a member of a union"
+
```

Comment:
Nit: usually don't want this whitespace after the docstring, unless it's a really big function where you're using whitespace to differentiate sections.